src/impl/regexParser.js

```diff
@@ -118,6 +118,29 @@ function extractIANAZone(match, cursor) {
   return [{}, zone, cursor + 1];
 }
 
+// ISO time parsing
+
+const isoTimeOnly = /^T?(\d\d)(?::?(\d\d)(?::?(\d\d)(?:[.,](\d{1,30}))?)?)?$/; 
```

Comment:
Aren't these identical to `isoTimeBaseRegex` and `extractISOTime`?

Reply:
They are very similar, but not quite identical.

* `isoTimeOnly` includes the optional `T` prefix, and matches the beginning/end of the string - `isoTimeBaseRegex` does not.

  This means `isoTimeBaseRegex` would match a value such as `foo00bar`, even though that's obviously not a valid Time value.
  But I guess we could consider changing it to this, to reuse the existing regex:

  ```
  const isoTimeOnly = RegExp(`^T?${isoTimeBaseRegex.source}$`)
  ```

* `extractISOTimeOnly` preserves the _precision_ of the value specified in the string, by only setting the values actually specified in the string - `isoTimeBaseRegex` always sets e.g. `seconds` to 0, even if not specified.

  > Either the seconds, or the minutes and seconds, may be omitted from the basic or extended time formats for greater brevity but decreased precision - https://en.wikipedia.org/wiki/ISO_8601#Times

  One could of course argue `extractISOTime` should be fixed so it too preserves the precision - I'm just afraid that might be a breaking change, as code may have been written to expect `seconds` in a `DateTime` instance to always have a value.

  There's also a minor difference in the casing of the property names in the returned object - e.g. `seconds` instead of `second`. That's because `Duration.fromObject` only accepts the plural version, which kinda makes sense, as it may represent arbitrary durations, and the singular name only makes sense when talking specifically about a time of day.